collated/main/charlesgoh.md

```diff
@@ -0,0 +1,632 @@
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        //sortBy then chooses the right ordering
+        switch (order) {
+        case "asc":
+            Collections.sort(internalList, comparator);
+            break;
+
+        case "desc":
+            Collections.sort(internalList, Collections.reverseOrder(comparator));
+            break;
+
+        default:
+            try {
+                System.out.println("An error occured");
+                throw new Exception("Invalid field parameter entered...\n");
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+    @Override
+    public Iterator<Person> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniquePersonList // instanceof handles nulls
+                        && this.internalList.equals(((UniquePersonList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+}
+```
+###### \java\seedu\address\storage\AddressBookStorage.java
+``` java
+    /**
+     * Saves the given {@link ReadOnlyAddressBook} to the storage.
+     * @param addressBook cannot be null.
+     * @throws IOException if there was any problem writing to the file.
+     */
+    void saveAddressBook(ReadOnlyAddressBook addressBook) throws IOException;
+
+    /**
+     * @see #saveAddressBook(ReadOnlyAddressBook)
+     */
+    void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) throws IOException;
+
+    void backupAddressBook(ReadOnlyAddressBook addressBook, String args) throws IOException;
+
+    void backupAddressBookDefault(ReadOnlyAddressBook addressBook) throws IOException;
+
+    void changeFilePath(String fp, UserPrefs u);
+    //@author
+}
+```
+###### \java\seedu\address\storage\StorageManager.java
+``` java
+    /**
+     * Backs up data to a remote location.
+     * @param addressBook
+     * @param filePath
+     * @throws IOException
+     */
+    public void backupAddressBook(ReadOnlyAddressBook addressBook, String filePath) throws IOException {
+        logger.fine("Attempting to write to backup data file in custom location");
+        this.saveAddressBook(addressBook, filePath);
+    }
+
+    /**
+     * Default back up data which saves file in the same directory as the main save file
+     * @param addressBook
+     * @throws IOException
+     */
+    public void backupAddressBookDefault(ReadOnlyAddressBook addressBook) throws IOException {
+        logger.fine("Attempting to write to backup data file");
+        this.saveAddressBook(addressBook, addressBookStorage.getAddressBookFilePath() + "-backup.xml");
+    }
+```
+###### \java\seedu\address\ui\MainWindow.java
+``` java
+    /**
+     * Handle increase font size command
```

Comment:
Coding standard violation:exclamation: The first statement of the header comment does not follow the prescribed style. See point 4, 3rd bullet point of the [[Java Coding Standard: Comments](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#comments)]